### PR TITLE
docs(pharos-site): add toggle button group documentation

### PR DESCRIPTION
**Pull request by Lham42**
**This change:** (check at least one)

- [ ] Adds a new feature
- [ ] Fixes a bug
- [ ] Improves maintainability
- [x] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [x] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [x] Code coverage maximal?
- [x] Changeset added?

**What does this change address?**
Adds documentation for the `PharosToggleButtonGroup`

### CODE REVIEW

File: packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import PageSection from '../../src/components/statics/PageSection.tsx';
+import BestPractices from '../../src/components/statics/BestPractices.tsx';
+
+<PageSection
+  isHeader
+  title="Toggle Button Group"
+  storyBookType="components"
+  description="Toggle button groups present related actions together to allow the user to select a mutually exclusive option."
```
Comment (daneah) [inline packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx:8]:
```suggestion
  description="Toggle button groups present related actions together to allow the user to select from mutually exclusive options."
```

File: packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+></PageSection>
+
+<PageSection title="Usage">
+  <p>
+    Within a toggle button group, only one option can be selected at the time. Selecting one option
```
Comment (daneah) [inline packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx:13]:
```suggestion
    Within a toggle button group, only one option can be selected at a time. Selecting one option
```

File: packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  </p>
+</PageSection>
+
+<PageSection title="Best practices">
+  <BestPractices
+    Do={
+      <ul>
+        <li>
+          Items within a toggle button group should be related and have a logical connection to each
+          other
```
Comment (daneah) [inline packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx:28]: This item lacks parallel structure with the others, which finish the sentence "Do ________". Maybe this would work? I think we don't have solid parallel structure for our dos/don'ts, so maybe I'm overthinking at this particular juncture.

```suggestion
          Group items within a toggle button group that are related and have a logical connection to each
          other
```

File: packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+          Place the toggle button group in close proximity (ideally just above) to the area it
+          controls
+        </li>
+      </ul>
+    }
+    Dont={
+      <ul>
+        <li>Don't group items that aren't clearly related</li>
+        <li>Don't use an icon-only toggle if the actions aren't clearly understood</li>
+        <li>Don't use the button group component if there are more than three options</li>
```
Comment (daneah) [inline packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx:45]:
```suggestion
        <li>Don't use a toggle button group if there are more than three options</li>
```

File: packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        <PharosLink href="https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html">
+          1.3.1 Info and Relationships A
+        </PharosLink>
+      </li>
+    </ul>
+  </PageSection>
+  <PageSection title="Code expectations" subSectionLevel={1}>
+    <ul>
+      <li>
+        There is a <code>role="group"</code> tag on the parent element <code>&lt;div&gt;</code>
```
Comment (daneah) [inline packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx:106]: Also: **must** the parent be a div?

```suggestion
        There is a <code>role="group"</code> attribute on the parent element <code>&lt;div&gt;</code>
```

File: packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+          </li>
+        </ul>
+      </li>
+      <li>
+        Use the semantic <code> &lt;button&gt; </code> on all grouped elements under the parent
+        element
+      </li>
+      <li>
+        If the button needs additional information in order to be understood by assistive
+        technologies, use an aria-label to convey this information
```
Comment (daneah) [inline packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx:119]: Either "use an `aria-label` attribute" or "use an ARIA label" might be better here.

File: packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx (hunk 5 above)
Comment (Lham42) [reply packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx:106]: Good point. I just followed the doc but we can change it

**Review by Niznikr — APPROVED** [review]

**Comment by Niznikr** [comment]
Could you also update the component status page?

File: packages/pharos-site/src/pages/components/component-status/status.json [hunk 7]
```diff
@@ -669,11 +669,11 @@
       "description": ""
     },
     "tests": {
-      "status": "planned",
+      "status": "released",
```
Comment (Niznikr) [inline packages/pharos-site/src/pages/components/component-status/status.json:672]:
```suggestion
      "status": "planned",
```
This refers to functional tests, which have not been made yet for this component.

File: packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx (hunk 5 above)
Comment (daneah) [reply packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx:106]: We could probably just shorten this to "...on the parent element" in that case!

**Comment by daneah** [comment]
@Lham42 while you're updating the status page—I believe Alert now has *released* rather than *legacy* styling as of #36

**Review by daneah — APPROVED** [review]
